### ISSUE TITLE
gvfs-no-bluray

### ISSUE THREAD

**Issue by yuyichao** [archlinuxcn/repo#53]
Any reason a package with different configuration and one patch version left behind should replace an official package?

**Comment by yuyichao** [comment]
@colinkeenan

**Comment by colinkeenan** [comment]
I originally put this in the AUR because gvfs sometimes thinks an internal
hard drive is a blue-ray drive for some reason. I tested a later version of
gvfs, and did not experience that issue anymore, and asked for this to be
removed from AUR. However, somebody objected, saying they still have this
error, so I kept it in the AUR. I pretty much put everything that was being
used by somebody that I had packaged for AUR into archlinuxcn/repo so that
I was not using anything directly from AUR.

Do you want me to remove this from archlinuxcn?

**Comment by yuyichao** [comment]
Oh, no. Keeping it here is perfectly fine and I am not trying to decide if the package is useful or now.

However, it seems that the package is replacing the official `gvfs` and is a little bit out-of-date. I'll be all good (for me) if it does not replace `gvfs` (and probably upgrade to the latest version).

**Comment by colinkeenan** [comment]
OK, I didn't notice it was out of date. I will look at that now.

**Comment by colinkeenan** [comment]
I've updated gvfs-no-bluray in AUR and archlinuxcn.

**Comment by colinkeenan** [comment]
How can it not replace gvfs though? It can't be installed along with gvfs.

**Comment by yuyichao** [comment]
You only need to `conflicts` `gvfs` not replacing it.

Conflicts means if someone want to install `gvfs-no-bluray` (e.g. because they have the problem you saw), `pacman` will automatically uninstall `gvfs`.

Replaces means if someone is using a binary repo that has `gvfs-no-bluray` in it (e.g. `archlinuxcn`), `pacman` will by default uninstall `gvfs` and install `gvfs-no-bluray` instead. If it is used in interactive mode, `pacman` will ask if you want to uninstall `gvfs`.

In short, `conflicts` make sure ppl that want to use it won't have the two package installed at the same time (which I think is what you want) while `replaces` will bother everyone else who has `gvfs` installed to replace it with `gvfs-no-bluray` even if they don't have the problem at all.

**Comment by colinkeenan** [comment]
OK, thanks for the explanation. I have removed 'gvfs' from the list of what this replaces and uploaded to AUR and archlinuxcn. Let me know if there's still a problem.

**Comment by yuyichao** [comment]
Looks good now. =)
